src/test/daml/Daml/Finance/Asset/Test/IbericoPigLifeCycle.daml

```diff
@@ -0,0 +1,134 @@
+  observations = M.fromList [(dateToDateClockTime (matureDate), 1.0)]
+  lifecycleObserver = [("PigLifecycler", S.singleton $ S.singleton farmer)]
+
+pigPriceOnMatureDateCid <- toInterfaceContractId  <$> submit butcher do 
+  createCmd Observation with provider = butcher; obsKey = show Pig; observations; observers = M.fromList lifecycleObserver
+-- butcherPigPriceOnMatureDateCid <- toInterfaceContractId  <$> submit foodProcessor do createCmd Observation with provider = foodProcessor; obsKey = show Butchered; observations; observers = M.fromList lifecycleObserver
+-- agedHamPriceOnMatureDateCid <- toInterfaceContractId  <$> submit restaurant do createCmd Observation with provider = restaurant; obsKey = show Aged; observations; observers = M.fromList lifecycleObserver
+-- restaurantHamPriceOnMatureDateCid <- toInterfaceContractId  <$> submit restaurant do createCmd Observation with provider = restaurant; obsKey = show Restaurant; observations; observers = M.fromList lifecycleObserver
+
+pigSoldEventCid <- toInterfaceContractId <$> submitMulti [farmer, butcher] [] do 
+  createCmd StateTransitionEvent with currentOwner = farmer; newOwner = butcher; eventTime; currentState = state
+
+-- create clock with mature to declare that current time is mature date
+clockCid <- toInterfaceContractId <$> submit farmer do 
+  createCmd DateClock with u = Unit matureDate; id = show matureDate; provider = farmer; observers = M.empty
+(soldPigCid , [effectCid]) <- Instrument.submitExerciseInterfaceByKeyCmd @Lifecyclable.I [farmer] [public] ibericoPigKey 
+  Lifecyclable.Lifecycle with settler; eventCid = pigSoldEventCid; 
+                                observableCids = [pigPriceOnMatureDateCid]; 
+                                ruleName = show state; clockCid
+
+-- Settlement, create settlement factory, settlement rule and get claim result                                                                                                                                                                   
+factoryCid <- submit farmer do createCmd BatchFactory with provider = farmer; observers = S.empty
+
+settlementRuleCid <- submitMulti [bank, farmer] [] do
+    createCmd Rule
+      with
+        custodian = bank
+        owner = farmer
+        claimers = S.singleton farmer -- Can I make new instrument go into butcher's account? No, not in the current rule implementation
+        settler
+        instructableCid = toInterfaceContractId factoryCid
+
+result <- submitMulti [farmer] [public] do
+  exerciseCmd settlementRuleCid SettlementRule.Claim with
+    claimer = farmer
+    holdingCids = [toInterfaceContractId @Holding.I transferableIbericoPig]
+    effectCid
+let 
+  Some [soldPigHoldingsCid] = result.newInstrumentHoldingCids
+  [paymentForPigInstructionCid] = result.instructionCids
+  paymentBatchForPigCid = result.containerCid
+
+-- can we setup instruction for butcher to pay farmer not for bank to pay farmer? No, not in the current rule implementation
```

Comment:
For farmer selling Pig to Butcher, the settlement instruction should really be butcher paying cash to farmer via bank, not bank to farmer directly. Yet I am not sure how to properly set up the instruction with current implementation of `SettlementRule`.

Reply:
This is indeed not possible currently (see my other comment) as lifecycle events are between custodian and owner.